File: week-05/Jukebox/Rock.cpp
#include <string>
#include "Rock.h"

using namespace std;

Rock::Rock(string title, string artist) : Song(title, artist) {
    genre = "Rock";
}

bool Rock::rate(unsigned int _rate) {
    if (_rate > 2 && _rate < 6) {
        ++count_of_ratings;
        sum_of_ratings += _rate;
        return true;
    } else {
        return false;
    }
}

File: week-12/Jukebox/Rock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JukeboxCS
{
    public class Rock : Song
    {
        public Rock(string mytitle, string myartist)
        {
            title = mytitle;
            artist = myartist;
            genre = "Rock";
        }

        public override bool Rate(int myrate)
        {
            if (myrate > 0 && myrate < 6)
            {
                ++ratecount;
                ratesum += myrate;
                return true;
            }
            else
            {
                return false;
            }
        }

    }
}
